crates/noirc_evaluator/src/ssa_refactor/ssa_gen/program.rs

```diff
@@ -17,10 +17,17 @@ pub(crate) struct Ssa {
 impl Ssa {
     /// Create a new Ssa object from the given SSA functions.
     /// The first function in this vector is expected to be the main function.
+    /// TODO: The above condition is somewhat hidden in the API, should we change it so that
+    /// TODO the caller needs to pass in the main function?
```

Comment:
Possibly, I'm not sure of a clean way to do so however. I think we'd be relying on users to manually extract main from a Vec only to have this function re-insert it afterward. The FunctionBuilder is the only real user of this function I believe so another option is to restrict the visibility a bit more.